Onboarding criada.

gabrielfbarbosa opened the pull request.

Commit:
primeiro modelo de onboarding

Commit:
adicionando model/arredondando botão

Commit:
corrigindo de OnboardingMain para OnboardingFragment

yanokumoto committed:
Criação tela de onboarding

Commit:
removendo nav_graph_onboarding e mantendo apenas nav_graph

gabrielfbarbosa committed:
Merge pull request #5 from gabrielfbarbosa/feature/onboarding

removendo nav_graph_onboarding e mantendo apenas nav_graph

yuukidesu9 committed:
Implementação inicial da tela principal

yuukidesu9 committed:
Tela principal alterada de FrameLayout para ConstraintLayout

yuukidesu9 committed:
Código alterado de Java para Kotlin

geanbrandao commented on app/src/main/java/br/com/compass/compassmart/ui/activity/MainActivity.kt:
```diff
@@ -8,5 +8,7 @@ class MainActivity : AppCompatActivity() {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+
+        supportActionBar?.hide() //Remove up app_bar
```
A toolbar no topo do app pode ser removida mudando um atributo no arquivo de theme, se ela nao é usada em nenhum lugar do app da pra remover direto lá.

gabrielfbarbosa replied:
Código removido da MainActivity. ActionBar removida diretamente no arquivo themes.xml

geanbrandao commented on app/src/main/java/br/com/compass/compassmart/ui/onboarding/OnboardingFragment.kt:
```diff
@@ -0,0 +1,86 @@
+
+    private val binding: FragmentOnboardingBinding by lazy {
+        FragmentOnboardingBinding.inflate(layoutInflater)
+    }
```
Aqui eu passei o implementação errada do viewBinding. O matheus me sinalizou que esse lazy pode dar algum problema com os ciclos de vida da fragment.

```suggestion
    private var _binding: FragmentOnboardingBinding? = null
    private val binding get() = _binding!!
```

gabrielfbarbosa replied:
Alterado.

geanbrandao commented on app/src/main/java/br/com/compass/compassmart/ui/onboarding/OnboardingFragment.kt:
```diff
@@ -0,0 +1,86 @@
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+
```
Precisa adicionar essa linha agora, por conta da nova implementação do binding.

```suggestion
_binding = ResultProfileBinding.inflate(inflater, container, false)
```

gabrielfbarbosa replied:
Adicionada.

geanbrandao commented on app/src/main/java/br/com/compass/compassmart/ui/onboarding/OnboardingFragment.kt:
```diff
@@ -0,0 +1,86 @@
+                super.onPageScrollStateChanged(state)
+                mudarCorBolinhaProgresso()
+            }
+        })
```
Pode extrair esse bloco de código como um função privada. 
por exemplo: 
`private fun setupViewPager()`

gabrielfbarbosa replied:
Bloco extraído e função criada.

geanbrandao commented on app/src/main/java/br/com/compass/compassmart/ui/onboarding/OnboardingFragment.kt:
```diff
@@ -0,0 +1,86 @@
+        return binding.root
+    }
+    // Adiciona os conteudos dos cards
+    fun adicionaNaLista(titulo : String, descricao : String, imagem: Int){
```
Função deve ser privada, pois só é usada dentro dessa fragment. Ficar atento aos modificadores de visibilidade
```suggestion
    private fun adicionaNaLista(titulo : String, descricao : String, imagem: Int) {
```

gabrielfbarbosa replied:
Alterado para função privada.

geanbrandao commented on app/src/main/java/br/com/compass/compassmart/ui/onboarding/OnboardingFragment.kt:
```diff
@@ -0,0 +1,86 @@
+        data.add(InformacoesModel(titulo, descricao, imagem))
+    }
+    // Conteudo de cada card
+    fun conteudo() {
```
```suggestion
    private fun conteudo() {
```

gabrielfbarbosa replied:
Alterado para função privada

geanbrandao commented on app/src/main/java/br/com/compass/compassmart/ui/onboarding/OnboardingFragment.kt:
```diff
@@ -0,0 +1,86 @@
+
+        adicionaNaLista("Tenha o poder que \n" + "você merece!",
+            "Comece a comprar agora mesmo\n" + "para mudar a sua vida!",
+            R.drawable.img_onboarding3)
```
Como essas strings são colocadas diretamente na tela e nao mudam, você precisa extrair elas para o arquivo de string resources

gabrielfbarbosa replied:
Strings extraídas.

geanbrandao commented on app/src/main/res/layout/fragment_onboarding.xml:
```diff
@@ -0,0 +1,97 @@
+        android:layout_below="@+id/onboarding_main_view_pager"
+        android:layout_centerHorizontal="true"
+        android:layout_marginBottom="40dp"
+        android:text="Ir para às compras!"
```
Extrair para o string resources

gabrielfbarbosa replied:
String extraída.

geanbrandao commented on app/src/main/java/br/com/compass/compassmart/TelaPrincipalFragment.java:
```diff
@@ -0,0 +1,61 @@
+    private String mParam1;
+    private String mParam2;
+
+    public TelaPrincipalFragment() {
+        // Required empty public constructor
+    }
+
+    /**
+     * Use this factory method to create a new instance of
+     * this fragment using the provided parameters.
+     *
+     * @param param1 Parameter 1.
+     * @param param2 Parameter 2.
+     * @return A new instance of fragment TelaPrincipalFragment.
+     */
+    public static TelaPrincipalFragment newInstance(String param1, String param2) {
+        TelaPrincipalFragment fragment = new TelaPrincipalFragment();
+        Bundle args = new Bundle();
+        args.putString(ARG_PARAM1, param1);
+        args.putString(ARG_PARAM2, param2);
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        if (getArguments() != null) {
+            mParam1 = getArguments().getString(ARG_PARAM1);
+            mParam2 = getArguments().getString(ARG_PARAM2);
+        }
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        return inflater.inflate(R.layout.fragment_tela_principal, container, false);
+    }
+}
```
Essa classe ta em JAVA. Devia ser em kotlin, ta perdida aqui no meio

gabrielfbarbosa replied:
Classe removida.

Commit:
corrigindo sugestões

Commit:
removendo ActionBar no themes

Commit:
corrigindo sugestões/removeno ActionBar no themes

Commit:
corrigindo sugestões/removendo ActionBar no themes

Commit:
Layout básico finalizado (ScrollView, ImageView, TextView e Button)

IsadoraPicoli committed:
criando tela de login

yuukidesu9 committed:
Iniciando código do layout da tela principal

Commit:
Ajustes PR onboarding

Commit:
Layout criado

Commit:
Atualizando o Detalhes do Produto

Commit:
Criando fragment, layout, item e data class dos Produtos

Commit:
Aplicando CardView, utilizando GridLayoutManager e ScrollView

Commit:
Criando o Adapter, imagens dos produtos e atualizando o código da Tela Produtos

Commit:
Criando os string arrays dos modelos dos produtos

Commit:
Atualizando arquivos da Tela de Produtos

Commit:
Tirando o sample data do layout e mudando o nav graph para testar a Tela de Produtos

Commit:
Commit inicial

Commit:
Carrinho de compras montado

Commit:
Merge branch 'feature/sprint2' of https://github.com/gabrielfbarbosa/ProjetoCompassMart into feature/detalhesDoProduto

Commit:
Criando o layout da Tela Detalhes do Produto com dados estáticos, adicionando drawables dos smartphones e arrumando os dados estáticos da Tela Carrinho de Compras

yanokumoto committed:
Criando a Tela Detalhes do Produto com dados estáticos, adicionando drawables dos smartphones e arrumando os dados estáticos da Tela Carrinho de Compras

Commit:
Arrumando arquivos da Tela Produtos

Commit:
Merge branch 'feature/sprint2' of https://github.com/gabrielfbarbosa/ProjetoCompassMart into feature/produtos

 Conflicts:
	app/src/main/java/br/com/compass/compassmart/ui/onboarding/OnboardingAdapter.kt
	app/src/main/java/br/com/compass/compassmart/ui/onboarding/OnboardingFragment.kt
	app/src/main/res/navigation/nav_graph.xml

Commit:
Criando Tela de Produtos com dados estáticos e adicionando ao navigation

yanokumoto committed:
Criando Tela de Produtos com dados estáticos e adicionando ao navigation

Commit:
Readicionando o InformacoesModel que sumiu

yuukidesu9 committed:
Cor do scroll horizontal alterada para design no Figma

Commit:
Corrigindo e aplicando corretamente o GridLayoutManager no ProdutosFragment

Commit:
Tirando um import desnecessário no ProdutosFragment

IsadoraPicoli committed:
Adicionando a lógica da tela de login

IsadoraPicoli committed:
Adicionando actions no nav_graph

IsadoraPicoli committed:
Merge telaLogin com sprint2

IsadoraPicoli committed:
reorganizando pastas das telas

IsadoraPicoli committed:
Alterando o diretório do package de cada arquivo e da navegação

IsadoraPicoli committed:
Construindo layout da tela de cadastro

IsadoraPicoli committed:
Definindo navegação da tela de cadastro

Commit:
Criando adapter, arrumando itens da tela

Commit:
Melhorando o item_carrinho_compra.xml

Commit:
Ajuste de detalhes dp item_carrinho_compra.xml

Commit:
Ajustando botoes do fragment_carrinho_compras.xml

Commit:
corrigindo conflitos para o PR

Commit:
Corrigindo conflitos para o PR

Commit:
Merge branch 'develop' into feature/carrinhocompras

# Conflicts:
#	app/src/main/java/br/com/compass/compassmart/ui/onboarding/OnboardingAdapter.kt
#	app/src/main/java/br/com/compass/compassmart/ui/onboarding/OnboardingFragment.kt

Commit:
Ajustando pacotes

Commit:
Ajustando pacotes

Commit:
Arrumando o View

Commit:
Finalizando o layout da Tela Produtos

Commit:
Puxando atualizações da develop

Commit:
Ajustes de conflitos

IsadoraPicoli committed:
Alterações para fazer merge com develop

IsadoraPicoli committed:
Alterando no nav graph, a tag startDestination

IsadoraPicoli committed:
Merge remote-tracking branch 'origin/develop' into feature/telaLogin

# Conflicts:
#	app/src/main/res/layout/activity_main.xml
#	app/src/main/res/navigation/nav_graph.xml

Commit:
Correções do Code Review #1

IsadoraPicoli committed:
Merge branch 'develop' of https://github.com/gabrielfbarbosa/ProjetoCompassMart into feature/telaLogin

 Conflicts:
	app/src/main/res/navigation/nav_graph.xml

IsadoraPicoli committed:
Alterações no arquivo nav-graph

Commit:
Correções Code Review #2 (último)

yanokumoto committed:
Finalizando Tela Produtos e atualizando a develop

Commit:
Fazendo navegação da Onboarding para Produtos

Commit:
Navegação entre todas as fragments funcionando

yanokumoto committed:
Navegação entre todas as fragments funcionando

Commit:
Layout criado

Commit:
Seleção do metodo de pagamento funcionando

Commit:
Merge branch 'feature/meiospagamentos' into sprint3

IsadoraPicoli committed:
Adicionando a navegação da tela de cadastro

IsadoraPicoli committed:
Alteração do id do botão cadstrar e do editText confirma senha

IsadoraPicoli committed:
Alteração no nome e valor de algumas strings da tela de parabenizacao

IsadoraPicoli committed:
Criando layout da tela de parabenização

IsadoraPicoli committed:
Criando fragment da tela de parabenização no nav-graph

IsadoraPicoli committed:
Alterando o layout dos edit texts da tela de login

IsadoraPicoli committed:
Alterando o layout dos edit texts da tela de cadastro

IsadoraPicoli committed:
Adicionando a navegação da tela de cadastro

IsadoraPicoli committed:
Padronizando a label dos fragments no arquivo nav-graph

IsadoraPicoli committed:
Alterando a label do fragment de parabenização no arquivo nav-graph

IsadoraPicoli committed:
Atualizando branch feature/telaLogin

IsadoraPicoli committed:
Merge branch 'sprint3' into feature/telaLogin

IsadoraPicoli committed:
Merge pull request #14 from gabrielfbarbosa/feature/telaLogin

Tela login concluída

IsadoraPicoli committed:
Update nav-graph

yuukidesu9 committed:
Merge branch 'sprint3' into feature/telaprincipal

# Conflicts:
#	app/src/main/res/layout/fragment_tela_principal.xml

yuukidesu9 committed:
Implementação inicial da tela principal

Commit:
Criando arquivo Fragment e Layout da Tela de Endereço

Commit:
Merge branch 'sprint3' of https://github.com/gabrielfbarbosa/ProjetoCompassMart into feature/endereco

Commit:
Finalizando Tela de Endereço

Commit:
Arrumando e finalizando o fluxo de navegação da Tela Onboarding -> Tela Produtos -> Tela Detalhes de Produto -> Tela Login -> Tela Carrinho de Compra -> Tela Endereço -> Tela Meio de Pagamento.

yanokumoto committed:
Tela de Endereço finalizada + Fluxo entre as telas corrigido para apresentação do dia 25/06

Fluxo de navegação finalizado no momento: Tela Onboarding -> Tela Produtos -> Tela Detalhes de Produto -> Tela Login -> Tela Carrinho de Compra -> Tela Endereço -> Tela Meio de Pagamento.

IsadoraPicoli committed:
Merge da branch feature/telaCadastro com a branch sprint3

IsadoraPicoli committed:
Merge branch 'sprint3' of https://github.com/gabrielfbarbosa/ProjetoCompassMart into feature/telaParabenizacao

 Conflicts:
	app/src/main/res/navigation/nav_graph.xml

IsadoraPicoli committed:
Removendo fragment duplicado do nav-graph

IsadoraPicoli committed:
Merge pull request #18 from gabrielfbarbosa/feature/telaParabenizacao

Feature/tela parabenizacao

yuukidesu9 committed:
Merge branch 'sprint3' into feature/telaprincipal

yanokumoto committed:
Tela Principal Finalizada

Commit:
Navegação alinhada

IsadoraPicoli committed:
Adicionando fragment da tela de parabenização no nav-graph

IsadoraPicoli committed:
Adicionando e editando algumas strings

IsadoraPicoli committed:
Adicionando e editando algumas strings

IsadoraPicoli committed:
Editando o estilo dos botões

IsadoraPicoli committed:
Editando o estilo dos botões

IsadoraPicoli committed:
Adicionando título nas telas de login e cadastro

IsadoraPicoli committed:
Adicionando título nas telas de login e cadastro

Commit:
Layout criado

Commit:
Ajustando projeto para telas menores

Commit:
Ajuste para tela menor

Commit:
Merge branch 'feature/carrinhocompras' into sprint3

# Conflicts:
#	app/src/main/java/br/com/compass/compassmart/ui/fragment/tela_carrinhocompras/CarrinhoComprasFragment.kt
#	app/src/main/java/br/com/compass/compassmart/ui/fragment/tela_onboarding/OnboardingAdapter.kt
#	app/src/main/java/br/com/compass/compassmart/ui/fragment/tela_onboarding/OnboardingFragment.kt
#	app/src/main/res/layout/fragment_carrinho_compras.xml
#	app/src/main/res/layout/item_carrinho_compra.xml
#	app/src/main/res/navigation/nav_graph.xml

Commit:
Merge remote-tracking branch 'origin/sprint3' into sprint3

Commit:
Ajustes para funcionamento em telas grandes e pequenas

Mateus94 committed:
Merge pull request #20 from gabrielfbarbosa/sprint3

Merge da branch sprint3 na branch develop
